espflash/src/connection/mod.rs

```diff
@@ -441,12 +484,20 @@ impl Connection {
                             RomErrorKind::from(response.error),
                         )))
                     } else {
-                        Ok(response.value)
-                    }
-                }
-                _ => {
-                    continue;
+                        // Check if the response is a Vector and strip header (first 8 bytes)
+                        // https://github.com/espressif/esptool/blob/release/v4/esptool/loader.py#L481: data = p[8:]
```

Comment:
Please include a shorthash in any URLs you include linking to specific lines. No need to reproduce the linked line in the comment, either.